### PR TITLE
feat: normalize paths to expand `~`, closes #307

### LINKED ISSUE #307: Feature Request: Support `~` in `Paths`
It'll be great if `Paths` can support `~`. I often have to type out my home directory manually.

Thanks for this plugin.

### PR DESCRIPTION
**Pull request by stevenxxiu**

### CODE REVIEW

File: lua/grug-far/engine/astgrep/getArgs.lua [hunk 1]
```diff
@@ -54,6 +54,9 @@ local function getArgs(inputs, options, extraArgs, blacklistedFlags, forceReplac
   if #inputs.paths > 0 then
     local paths = utils.splitPaths(inputs.paths)
     for _, path in ipairs(paths) do
+      if not vim.startswith(path, '.') then
```
Comment (MagicDuck) [inline lua/grug-far/engine/astgrep/getArgs.lua:57]: fix looks good, but I would just extract a new function in `utils.lua` that gets used in both places:
```lua
function M.normalizePath(path) 
      if vim.startswith(path, '.') then
         return path        
      end
      return vim.fs.normalize(path)
end

```

**Review by MagicDuck — CHANGES_REQUESTED** [review]

**Comment by MagicDuck** [comment]
just popped into my mind, but could you also add an "example" in `defaultOptions.engines.ripgrep.placeholders.paths` (https://github.com/MagicDuck/grug-far.nvim/blob/9a2f78219390b47d67795ab09390d7f092e23976/lua/grug-far/opts.lua#L66) and `defaultOptions.engines.astgrep.placeholders.paths` (https://github.com/MagicDuck/grug-far.nvim/blob/9a2f78219390b47d67795ab09390d7f092e23976/lua/grug-far/opts.lua#L89) in `opts.lua` so that people know this feature is available?
for instance we could change it to:
```lua
        paths = 'ex: /foo/bar   ../   ./hello\\ world/   ./src/foo.lua  ~/.config',
```
You might have to update the test screenshots after that change, see: https://github.com/MagicDuck/grug-far.nvim/blob/main/CONTRIBUTING.md
likely you only need to run:
```sh
make test update_screenshots=true
```

**Comment by stevenxxiu** [comment]
Thanks for the review. I addressed your comments above.

The *ast-grep* version seems a bit out-of-date so my screenshots errored on *GitHub* at first.

**Review by MagicDuck — APPROVED** [review]

**Comment by MagicDuck** [comment]
> The ast-grep version seems a bit out-of-date so my screenshots errored on GitHub at first.
oh interesting, I did not consider this. Will need to update the Makefile to fetch the same version that is used in CI!

**Pull request merged**